tools/tck-regression/src/main/java/org/opencypher/tools/tck/regression/TckRegressionModel.java

```diff
@@ -0,0 +1,233 @@
+        private final List<Scenario> newlyPassingScenarios;
+        private final List<Scenario> newlyFailedScenarios;
+        private final List<Scenario> allScenarios;
+        private final int totalPassingScenarios;
+        private final int totalScenarios;
+
+        String passingPercentage;
+
+        Diff(List<Scenario> newlyPassingScenarios, List<Scenario> newlyFailedScenarios,
+             int totalPassingScenarios, int totalScenarios, List<Scenario> allScenarios) {
+            this.newlyPassingScenarios = unmodifiableList(newlyPassingScenarios);
+            this.newlyFailedScenarios = unmodifiableList(newlyFailedScenarios);
+            this.allScenarios = unmodifiableList(allScenarios);
+            this.totalPassingScenarios = totalPassingScenarios;
+            this.totalScenarios = totalScenarios;
+            this.passingPercentage = new DecimalFormat("#.##").format((0.0 + totalPassingScenarios) / totalScenarios * 100) + "%";
+        }
+
+        public List<Scenario> getNewlyPassingScenarios() {
+            return newlyPassingScenarios;
+        }
+
+        public List<Scenario> getNewlyFailedScenarios() {
+            return newlyFailedScenarios;
+        }
+
+        public int getTotalPassingScenarios() {
+            return totalPassingScenarios;
+        }
+
+        public int getTotalScenarios() {
+            return totalScenarios;
+        }
+
+        public String getPassingPercentage() {
+            return passingPercentage;
+        }
+
+        public List<Scenario> getAllScenarios() {
+            return allScenarios;
+        }
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @JacksonXmlRootElement(localName = "testsuite")
+    static class TestSuite {
+        @JacksonXmlElementWrapper(localName = "testcase", useWrapping = false)
+        private final List<Scenario> testcase;
+
+        @JsonCreator
+        public TestSuite(@JacksonXmlProperty(localName = "testcase") List<Scenario> testcase) {
+            this.testcase = testcase;
+        }
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    static class Failure {
+    }
+
+    @SuppressWarnings("unused")
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Scenario {
+        private static Pattern pattern = Pattern.compile("Feature \"(.*?)\": Scenario \"(.*?)\"");
+
+        private final String name;
+        private final String featureName;
+        private final boolean failed;
+
+        public Scenario(String featureName, String name, boolean failed) {
+            this.name = name;
+            this.featureName = featureName;
+            this.failed = failed;
+        }
+
+        @JsonCreator
+        public static Scenario create(@JsonProperty("name") String title, @JsonProperty("failure") Failure failure) {
+            Matcher matcher = pattern.matcher(title);
+            checkState(matcher.find(), "Expecting string in format `Feature [name]: Scenario: [name]`, got `%s`", title);
```

Comment:
`Scenario.toString()` is [Feature "[name]": Scenario "[name]"](https://github.com/opencypher/openCypher/blob/master/tools/tck-api/src/main/scala/org/opencypher/tools/tck/api/Scenario.scala#L47). This is working in Cypher for Gremlin because of different JUnit version. There seems to be a bug in [newer JUnit](https://github.com/gradle/gradle/issues/5975) when DynamicTest displayName is ignored. I will look into this. Will look for workaround, or use [cucumber .json](https://github.com/opencypher/openCypher/pull/360) as source.

Reply:
There is a colon too much in the error message, after "Scenario", and the quotes around `[name]` are missing.

Reply:
Thanks for looking into a workaround.

Reply:
DynamicTest displayName in report was available before JUnit 5.1.1 version. This will be [fixed in Surefire 3.0.0-M4](https://issues.apache.org/jira/browse/SUREFIRE-1546) (not released yet).

I've added support for Cucumber reports.

So comparison tool will be useful only after Surefire release or Cucumber reporting support in openCypher. Could you please take a look at [this PR](https://github.com/opencypher/openCypher/pull/360)?